lib/web/apiserver.go

```diff
@@ -3169,9 +3169,14 @@ func (h *Handler) clusterUnifiedResourcesGet(w http.ResponseWriter, request *htt
 				r.GetApp().GetName(): allowedAWSRoles,
 			}
 
+			proxyDNSName := h.proxyDNSName()
+			if r.GetApp().GetUseAnyProxyPublicAddr() {
+				proxyDNSName = utils.FindMatchingProxyDNS(request.Host, h.proxyDNSNames())
+			}
```

Comment:
nit: add comment why `FindMatchingProxyDNS` needs to happen here. Please adjust comment as needed. 
```suggestion
			proxyDNSName := h.proxyDNSName()
			if r.GetApp().GetUseAnyProxyPublicAddr() {
			        // let the current proxy user is connected to override the dns name
				proxyDNSName = utils.FindMatchingProxyDNS(request.Host, h.proxyDNSNames())
			}
```